source/MaterialXTest/ShaderValid.cpp

```diff
@@ -1282,25 +1283,41 @@ TEST_CASE("MaterialX documents", "[shadervalid]")
     mx::CopyOptions importOptions;
     importOptions.skipDuplicateElements = true;
 
-#if defined(MATERIALX_BUILD_GEN_GLSL) || defined(MATERIALX_BUILD_GEN_OGSFX)
```

Comment:
Fix and old bug. Split to separate light handling for GLSL vs OGSFX. The glslShaderGenerator was always referenced below and would crash if GLSL tests where disabled.